src/ui/api/projectmember.go

```diff
@@ -179,18 +210,27 @@ func AddOrUpdateProjectMember(projectID int64, request models.MemberReq) (int, e
 		}
 		member.EntityID = userID
 	} else if len(request.MemberGroup.LdapGroupDN) > 0 {
-		member.EntityType = common.GroupMember
-		//If groupname provided, use the provided groupname
-		//If ldap group already exist in harbor, use the previous group name
+		//Check duplicate LDAP DN in usergroup, if usergroup exist, return error
+		userGroupList, err := group.QueryUserGroup(models.UserGroup{LdapGroupDN: request.MemberGroup.LdapGroupDN})
+		if err != nil {
+			return 0, err
+		}
+		if len(userGroupList) > 0 {
+			return 0, ErrDuplicateLDAPGroup
+		}
+		//If groupname provided, use the provided groupname to name this group
 		groupID, err := auth.SearchAndOnBoardGroup(request.MemberGroup.LdapGroupDN, request.MemberGroup.GroupName)
```

Comment:
I think we should fail in `SearchAndOnBoard` or let SearchAndOnBoard in ldap package ensure there's no duplicate entry?

Reply:
OK